README.md
# sfst_helper

SFST using Flutter

## Getting Started

For help getting started with Flutter, view our online
[documentation](https://flutter.io/).


changelog.md
# Changelog
All notable changes to this project will be documented in this file.


## [Unreleased]
- Prepare for use of SharedPrefs to save checkbox value at close
- Officer's section
- Change implementation of about screen


## [2.1] - 2018-08-01
### Clues page
- Added case number, offender, and time and date

## [2.0] - 2018-07-19
### Clues page
- Ability to create a pdf report

## [1.7] - 2018-07-18
### Clues Page
    -   Rebuilt page list

## [1.6] - 2018-07-16
### Added
-Light and Dark toggle
-Changelog Webview

## [1.5] - 2018-07-07
### Added
- Rebuilt app using Flutter
- Changed primary color to a lighter blue

#### Main Page
- Switched from bottom navigation bar to tab view to switch between instructions
- Added FAB to go to clues pages
- Added icon button to jump to about page in top right corner
- Added ability to swipe between instruction pages
- Increased font size

#### Clues Page
- Created clues page
- Checkbox list to clues

#### About Page
- Enlarged font
- Added coded changelog to page




## [1.0.0] - 2018-06-29
### Added
- Initial build


test/widget_test.dart
// This is a basic Flutter widget test.
// To perform an interaction with a widget in your test, use the WidgetTester utility that Flutter
// provides. For example, you can send tap and scroll gestures. You can also use WidgetTester to
// find child widgets in the widget tree, read text, and verify that the values of widget properties
// are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:sfst_helper/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(new FirstScreen());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/main.dart
import 'package:dynamic_theme/dynamic_theme.dart';
import 'package:pdfview/pdfview.dart';
import 'package:flutter/material.dart';
import 'package:flutter_webview_plugin/flutter_webview_plugin.dart';


import 'about.dart';
import 'secondscreen.dart';

void main() {
  runApp(new MyApp());
}

final flutterWebviewPlugin = new FlutterWebviewPlugin();

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    // TODO: implement build
    return new DynamicTheme(
        defaultBrightness: Brightness.light,
        data: (brightness) => new ThemeData(
      primarySwatch: Colors.blue,
      accentColor: Colors.blueAccent,
      fontFamily: "Raleway",
      primaryColorDark: Colors.blue,
      brightness: brightness,
    ),
    themedWidgetBuilder: (context, theme) {
    return new MaterialApp(
    theme: theme,


        routes:
    {
    '/': (context) => FirstScreen(),
      '/about': (context) => AboutScreen(),
    '/c': (_) => new WebviewScaffold(
    url: "https://zavon.org/sfst/changelog.md",
    appBar: new AppBar(
    title: new Text("Changelog"),
    ),
    )
    });
    }
    );

  }
}





class FirstScreen extends StatefulWidget {
  @override
  FirstScreenState createState() => new FirstScreenState();
}

class FirstScreenState extends State<FirstScreen> {
  @override
  Widget build(BuildContext context) {

            return DefaultTabController(
              length: 3,
              child: Scaffold(
                  appBar: AppBar(
                    actions: <Widget>[
                      new IconButton(
                          icon: Icon(Icons.brightness_6),
                          onPressed: changeBrightness),
                      IconButton(
                          icon: Icon(Icons.info_outline),
                          onPressed: () {
                            Navigator.push(
                                context,
                                MaterialPageRoute(
                                    builder: (context) => AboutScreen()));
                          }),

    ],
                    bottom: TabBar(
                      tabs: [
                        Tab(text: "HGN"),
                        Tab(text: "WAT"),
                        Tab(
                          text: "OLS",
                        ),
                      ],
                    ),
                    title: Text('SFST Guide'),
                  ),
                  body: TabBarView(
                    children: <Widget>[
                      Container(
                        padding: new EdgeInsets.only(
                            top: 10.0, left: 8.0, right: 8.0),
                        child: ListView(
                          children: [
                            Container(
                              child: Text(
                                'Horizontal Gaze Nystagmus',
                                    textAlign: TextAlign.center,
                                    style: TextStyle(
                                      fontSize: 28.0,
                                      fontFamily: 'Montserrat'
                                    )
                              ),

                            ),
                            Text(
                              '\n1. I am going to check your eyes.\n\n2. Keep your head still and follow the stimulus with your eyes only.\n\n3. Do not move your head.\n\n4. Do you understand the instructions?',
                            textAlign: TextAlign.center,
                              style: TextStyle(
                                fontSize: 20.0,
                                fontFamily: 'Raleway'
                              ),
                            )
                          ],
                        )
                      ),
                      new Container(
                        padding: new EdgeInsets.only(
                            top: 10.0, left: 8.0, right: 8.0),
                          child: ListView(
                            children: [
                              Container(
                                child: Text(
                                    'Walk and Turn',
                                    textAlign: TextAlign.center,
                                    style: TextStyle(
                                        fontSize: 28.0,
                                        fontFamily: 'Montserrat'
                                    )
                                ),

                              ),
                              Text(
                                '\n1. Put your left foot on the line and put your right foot in front of it with your right heel touching your left toe. Keep your hands at your side.\n\n2.Do not start until I tell you to.\n\n3. Do you understand the directions?\n\n4. When I tell you to begin, take nine heel to toe steps on the line, turn around keeping one foot on the line, and return nine heel to toe steps.\n\n5. On the ninth step, keep the front foot on the line and turn by taking several small steps with the other foot.\n\n6. Whilewalking, watch your feet at all times, keep arms at side, count steps out loud. Once you begin, do not stop until test is completed.\n\n7. Do you understand the instructions?\n\n8. You may begin the test.\n',
                                textAlign: TextAlign.center,
                                style: TextStyle(
                                  fontSize: 20.0,
                                ),
                              )
                            ],
                          )
                      ),
                      new Container(
                          padding: new EdgeInsets.only(
                              top: 10.0, left: 8.0, right: 8.0),
                          child: ListView(
                            children: [
                              Container(
                                child: Text(
                                    'One Legged Stand',
                                    textAlign: TextAlign.center,
                                    style: TextStyle(
                                        fontSize: 28.0,
                                        fontFamily: 'Montserrat'
                                    )
                                ),

                              ),
                              Text(
                                '\n1. Stand with your heels together and your arms at your side.\n\n2. Do not begin until I tell you to.\n\n3. Do you understand?\n\n4. When I tell you, I want you to raise one leg, either leg, approximately six inches off the ground, foot pointed out. Keep both legs straight and keep your eyes on the elevated foot.\n\n5. While holding that position, count out loud; one thousand and one, one thousand and two, one thousand and three, and so forth until told to stop.\n\n6. Do you understand the instructions?\n\n7. You may begin the test.',
                                textAlign: TextAlign.center,
                                style: TextStyle(
                                  fontSize: 20.0,
                                ),
                              )
                            ],
                          )
                      ),
                    ],
                  ),
                  floatingActionButton: new FloatingActionButton(
                    child: new Icon(Icons.assignment),
                    onPressed: () {
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (BuildContext context) {
                          return new
                          SecondScreen();
                        }
                        ),
                      );
                    },
                  )),

          );
        }


  void changeBrightness() {
    DynamicTheme.of(context).setBrightness(Theme.of(context).brightness == Brightness.dark? Brightness.light: Brightness.dark);
  }
  void changeColor() {
    DynamicTheme.of(context).setThemeData(new ThemeData(
        primaryColor: Theme.of(context).primaryColor == Colors.indigo? Colors.red: Colors.indigo
    ));
  }
  }



lib/secondscreen.dart
import 'package:pdfview/pdfview.dart';
import 'package:flutter/material.dart';
import 'dart:async';
import 'package:intl/intl.dart';
import 'package:datetime_picker_formfield/datetime_picker_formfield.dart';
import 'package:datetime_picker_formfield/time_picker_formfield.dart';

class SecondScreen extends StatefulWidget {
  @override
  SecondScreenState createState() => SecondScreenState();
}

class SecondScreenState extends State<SecondScreen> {
//HGN
  bool hgn1 = false; //Lack of Smooth Pursuit
  bool hgn2 = false; //Distinct and Sustained Nystagmus at Maximum Deviation
  bool hgn3 = false; //Onset prior to 45 degrees
  bool hgn4 = false; //Lack of Smooth Pursuit
  bool hgn5 = false; //Distinct and Sustained Nystagmus at Maximum Deviation
  bool hgn6 = false; //Onset prior to 45 degrees

  String _hstring1;
  String _hstring2;
  String _hstring3;
  String _hstring4;
  String _hstring5;
  String _hstring6;

  //WAT
  bool wat1 = false; //Can't keep balance
  bool wat2 = false; //Starts too soon
  bool wat3 = false; //Stops Walking
  bool wat4 = false; //Misses heel to toe
  bool wat5 = false; //Steps off line
  bool wat6 = false; //Uses arms for balance
  bool wat7 = false; //Wrong number of steps
  bool wat8 = false; //Improper turn

  String _wstring1;
  String _wstring2;
  String _wstring3;
  String _wstring4;
  String _wstring5;
  String _wstring6;
  String _wstring7;
  String _wstring8;

  //OLS
  bool ols1 = false; //Sways
  bool ols2 = false; //Uses Arms for Balance
  bool ols3 = false; //Hops
  bool ols4 = false; //Puts Foot Down

  String _ostring1;
  String _ostring2;
  String _ostring3;
  String _ostring4;

  //Case Info
  final myController = TextEditingController();
  final offenderController = TextEditingController();

  String _caseNumber;
  String _date;
 String _offender;

  final dateFormat = DateFormat("EEEE, MMMM d, yyyy 'at' h:mm a");

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: new AppBar(
        title: new Text('Notes'),
      ),
      body: Container(
        child: new ListView(
          children: <Widget>[
            new Card(
              child: new Column(
                mainAxisSize: MainAxisSize.min,
                children: <Widget>[
                  Text(
                    'Case Information',
                    style:
                        TextStyle(fontSize: 18.0, fontWeight: FontWeight.bold),
                  ),
                  new ListTile(
                    leading: Icon(Icons.assignment),
                    title:       TextField(
                      decoration: InputDecoration(
                        hintText: 'Report Number',
                      ),
                      controller: myController,
                    ),

                  ),

                  new ListTile(
                    leading: Icon(Icons.person),
                    title:       TextField(
                      decoration: InputDecoration(
                        hintText: 'Offender',
                      ),
                      controller: offenderController,
                    ),

                  ),

                  new ListTile(
                    leading: Icon(Icons.today),
                    title: DateTimePickerFormField(
                        decoration: (InputDecoration(
                            hintText: 'Time and Date')),
                        format: dateFormat,
                        onChanged: (date) {
                          setState(() {
                            _date = dateFormat.format(date);
                          });
                        }),

                  ),

                ],
              ),
            ),
            new Card(
                child: new Column(
                    mainAxisSize: MainAxisSize.min,
                    children: <Widget>[
                  Text(
                    'Horizontal Gaze Nystagmus Left Eye',
                    style:
                        TextStyle(fontSize: 18.0, fontWeight: FontWeight.bold),
                  ),
                  new CheckboxListTile(
                    title: const Text('Lack of Smooth Pursuit'),
                    value: hgn1,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        hgn1 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text(
                        'Distinct and Sustained  Nystagmus at Maximum Deviation'),
                    value: hgn2,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        hgn2 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Onset Prior to 45 Degrees'),
                    value: hgn3,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        hgn3 = value;
                      });
                    },
                  ),
                ])),
            new Card(
                child: new Column(
                    mainAxisSize: MainAxisSize.min,
                    children: <Widget>[
                  Text(
                    'Horizontal Gaze Nystagmus Right Eye',
                    style:
                        TextStyle(fontSize: 18.0, fontWeight: FontWeight.bold),
                  ),
                  new CheckboxListTile(
                    title: const Text('Lack of Smooth Pursuit'),
                    value: hgn4,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        hgn4 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text(
                        'Distinct and Sustained  Nystagmus at Maximum Deviation'),
                    value: hgn5,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        hgn5 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Onset Prior to 45 Degrees'),
                    value: hgn6,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        hgn6 = value;
                      });
                    },
                  ),
                ])),
            new Card(
                child: new Column(
                    mainAxisSize: MainAxisSize.min,
                    children: <Widget>[
                  Text(
                    'Walk and Turn',
                    style:
                        TextStyle(fontSize: 18.0, fontWeight: FontWeight.bold),
                  ),
                  new CheckboxListTile(
                    title: const Text("Can't Keep Balance"),
                    subtitle: const Text('Instruction Stage'),
                    value: wat1,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat1 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Starts too Soon'),
                    subtitle: const Text('Instruction Stage'),
                    value: wat2,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat2 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text("Stops Walking"),
                    value: wat3,
                    onChanged: (bool value) {
                      setState(() {
                        wat3 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Misses Heel to Toe'),
                    value: wat4,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat4 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text("Steps Off Line"),
                    value: wat5,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat5 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Uses Arms for Balance'),
                    value: wat6,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat6 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text("Wrong Number of Steps"),
                    value: wat7,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat7 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Improper Turn'),
                    value: wat8,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        wat8 = value;
                      });
                    },
                  ),
                ])),
            new Card(
                child: new Column(
                    mainAxisSize: MainAxisSize.min,
                    children: <Widget>[
                  Text(
                    'One Legged Stand',
                    style:
                        TextStyle(fontSize: 18.0, fontWeight: FontWeight.bold),
                  ),
                  new CheckboxListTile(
                    title: const Text("Sways"),
                    value: ols1,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        ols1 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Uses Arms for Balance'),
                    value: ols2,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        ols2 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text("Hops"),
                    value: ols3,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        ols3 = value;
                      });
                    },
                  ),
                  new CheckboxListTile(
                    title: const Text('Puts Foot Down'),
                    value: ols4,
                    activeColor: Colors.blueAccent,
                    onChanged: (bool value) {
                      setState(() {
                        ols4 = value;
                      });
                    },
                  ),
                ])),
          ],
        ),
      ),
      floatingActionButton: new FloatingActionButton(
          child: new Icon(Icons.receipt),
          onPressed: () {
            _Shstring1();
            _Shstring2();
            _Shstring3();
            _Shstring4();
            _Shstring5();
            _Shstring6();
            _Swstring1();
            _Swstring2();
            _Swstring3();
            _Swstring4();
            _Swstring5();
            _Swstring6();
            _Swstring7();
            _Swstring8();
            _Sostring1();
            _Sostring2();
            _Sostring3();
            _Sostring4();
            _Soffender();
            _Scase();
            pdfview(
                """<!DOCTYPE html>
<html>
<head>
<style>
table {
    font-family: arial, sans-serif;
    border-collapse: collapse;
    width: 100%;
}

td, th {
    border: 1px solid #dddddd;
    text-align: left;
    padding: 8px;
}

tr:nth-child(even) {
    background-color: #dddddd;
}
</style>
</head>
<body>

<h1>SFST Notes</h1>
<h2>Case Number: $_caseNumber<hr>
Offender Name: $_offender <hr>
Date: $_date </h2>   

<h3>Horizontal Gaze Nystagmus</h3>
<table>
	<tr>
    	<td>Clue</td>
        <td>Left Eye</td>
        <td>Right Eye</td>
    </tr>
    <tr>
    	<td>Lack of Smooth Pursuit</td>
        <td>$_hstring1</td>
        <td>$_hstring4</td>
     </tr>
         <tr>
    	<td>Distinct and Sustained Nystagmus at Maximum Deviation</td>
        <td>$_hstring2</td>
        <td>$_hstring5</td>
     </tr>
              <tr>
    	<td>Onset of Nystagmus prior to 45 degrees</td>
        <td>$_hstring3</td>
        <td>$_hstring6</td>
     </tr>
     </table>
     
<h3>Walk and Turn</h3>
<table>
	<tr>
    	<td>Clue</td>
        <td>Displayed</td>
    </tr>
    <tr>
    	<td>Could not keep balance</td>
        <td>$_wstring1</td>
     </tr>
         <tr>
    	<td>Started too soon</td>
        <td>$_wstring2</td>
     </tr>
         	<td>Stopped Walking</td>
        <td>$_wstring3</td>
     </tr>
         <tr>
    	<td>Missed heel to toe</td>
        <td>$_wstring4</td>
     </tr>
         <tr>
    	<td>Stepped off line</td>
        <td>$_wstring5</td>
     </tr>
         <tr>
    	<td>Used arms for balance</td>
        <td>$_wstring6</td>
     </tr>
         	<td>Wrong number of steps</td>
        <td>$_wstring7</td>
     </tr>
         <tr>
    	<td>Improper turn</td>
        <td>$_wstring8</td>
     </tr>

     </table>
     
     <h3>One Legged Stand</h3>
<table>
	<tr>
    	<td>Clue</td>
        <td>Displayed</td>
    </tr>
    <tr>
    	<td>Swayed</td>
        <td>$_ostring1</td>
     </tr>
         <tr>
    	<td>Used arms for balance</td>
        <td>$_ostring2</td>
     </tr>
         	<td>Hopped</td>
        <td>$_ostring3</td>
     </tr>
         <tr>
    	<td>Put foot down</td>
        <td>$_ostring4</td>
     </tr>
      
        







</body>
</html> """);
          }),
    );
  }

void _Scase() {
    if (myController.text != null) {
      _caseNumber = myController.text;
    } else {
      _caseNumber = " ";
    }
}

 void _Soffender() {
    if (offenderController.text != null) {
      _offender = offenderController.text;
    } else {
      _offender = " ";
    }
  }



  void _Shstring1() {
    if (hgn1 == true) {
      _hstring1 =
          "&#9746;";
    } else {
      _hstring1 =
          "&#9744; ";
    }
    ;
  }

  void _Shstring2() {
    if (hgn2 == true) {
      _hstring2 =
          "&#9746;";
    } else {
      _hstring2 =
          "&#9744; ";
    }
    ;
  }

  void _Shstring3() {
    if (hgn3 == true) {
      _hstring3 =
          "&#9746;";
    } else {
      _hstring3 =
          "&#9744;";
    }
    ;
  }

  void _Shstring4() {
    if (hgn4 == true) {
      _hstring4 =
          "&#9746;";
    } else {
      _hstring4 =
          "&#9744;";
    }
    ;
  }

  void _Shstring5() {
    if (hgn5 == true) {
      _hstring5 =
          "&#9746;";
    } else {
      _hstring5 =
          "&#9744;";
    }
    ;
  }

  void _Shstring6() {
    if (hgn6 == true) {
      _hstring6 =
          "&#9746;";
    } else {
      _hstring6 =
          "&#9744;";
    }
    ;
  }

  void _Swstring1() {
    if (wat1 == true) {
      _wstring1 = "&#9746;";
    } else {
      _wstring1 = "&#9744;";
    }
    ;
  }

  void _Swstring2() {
    if (wat2 == true) {
      _wstring2 = "&#9746;";
    } else {
      _wstring2 = "&#9744;";
    }
    ;
  }

  void _Swstring3() {
    if (wat3 == true) {
      _wstring3 = "&#9746;";
    } else {
      _wstring3 = "&#9744;";
    }
    ;
  }

  void _Swstring4() {
    if (wat4 == true) {
      _wstring4 = "&#9746;";
    } else {
      _wstring4 = "&#9744;";
    }
    ;
  }

  void _Swstring5() {
    if (wat5 == true) {
      _wstring5 = "&#9746;";
    } else {
      _wstring5 = "&#9744;";
    }
    ;
  }

  void _Swstring6() {
    if (wat6 == true) {
      _wstring6 = "&#9746;";
    } else {
      _wstring6 = "&#9744;";
    }
    ;
  }

  void _Swstring7() {
    if (wat7 == true) {
      _wstring7 = "&#9746;";
    } else {
      _wstring7 = "&#9744;";
    }
  }

  void _Swstring8() {
    if (wat8 == true) {
      _wstring8 = "&#9746;";
    } else {
      _wstring8 = "&#9744;";
    }
  }

  void _Sostring1() {
    if (ols1 == true) {
      _ostring1 = "&#9746;";
    } else {
      _ostring1 = "&#9744;";
    }
  }

  void _Sostring2() {
    if (ols2 == true) {
      _ostring2 = "&#9746;";
    } else {
      _ostring2 = "&#9744;";
    }
  }

  void _Sostring3() {
    if (ols3 == true) {
      _ostring3 = "&#9746;";
    } else {
      _ostring3 = "&#9744;";
    }
  }

  void _Sostring4() {
    if (ols4 == true) {
      _ostring4 = "&#9746;";
    } else {
      _ostring4 = "&#9744;";
    }
  }
}


lib/about.dart
import 'package:flutter/material.dart';
import 'package:flutter_webview_plugin/flutter_webview_plugin.dart';

import 'main.dart';


class AboutScreen extends StatefulWidget {
  @override
  AboutScreenState createState() => new AboutScreenState();
}

class AboutScreenState extends State<AboutScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: new AppBar(
          title: new Text("About"),
        ),
        body: Container(
          child: ListView(
            children: <Widget>[
              new Text(
                'Version\n',
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 28.0
                ),
              ),
              Text(
                '2.1\n',
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 20.0
                ),
              ),
              Text(
                'Developer\n',
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 28.0
                ),
              ),
              Text(
                'Zavon\n',
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 20.0
                ),
              ),
              Text(
                'Contact\n',
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 28.0
                ),
              ),
              Text(
                'zavon@zavon.org',
                textAlign: TextAlign.center,
                style: TextStyle(
                  fontSize: 20.0,
                ),
              )
            ],
          ),
        ),

        // ...


        floatingActionButton: new FloatingActionButton(
          child: new Icon(Icons.announcement),
          onPressed: () {
            Navigator.pushNamed(context, '/c');
          },
        ));
  }
}